File: app/src/main/java/com/madebyaron/betterhealth/view/list/dialogs/AddCigarettesDialog.java
package com.madebyaron.betterhealth.view.list.dialogs;

import android.app.Dialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.text.Editable;
import android.view.LayoutInflater;

import androidx.fragment.app.FragmentActivity;
import androidx.lifecycle.ViewModelProviders;

import com.madebyaron.betterhealth.databinding.DialogAddCigarettesBinding;
import com.madebyaron.betterhealth.view.components.DateTimePicker;
import com.madebyaron.betterhealth.viewmodel.ListModel;

import java.util.Date;

public class AddCigarettesDialog extends Dialog {

    private final FragmentActivity activity;

    public AddCigarettesDialog(FragmentActivity activity) {
        super(activity);
        this.activity = activity;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        DialogAddCigarettesBinding binding = DialogAddCigarettesBinding.inflate(LayoutInflater.from(getContext()));
        setContentView(binding.getRoot());

        if (getWindow() != null) {
            getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        }

        ListModel model = ViewModelProviders.of(activity).get(ListModel.class);
        Date date = new Date();

        DateTimePicker.setDateTimePickerText(binding.addCigarettesDateInput, date);
        binding.addCigarettesDateInput.setOnClickListener(view ->
                DateTimePicker.openDateTimePickerDialog(getContext(), date, dateTime -> {
                            date.setTime(dateTime.getTime());
                            DateTimePicker.setDateTimePickerText(binding.addCigarettesDateInput, date);
                        }
                )
        );

        binding.addCigarettesAddButton.setOnClickListener(view -> {
            dismiss();
            Editable countInput = binding.addCigarettesCountInput.getText();
            String countString = countInput == null ? "" : countInput.toString();
            model.addCigarettes(date, countString.isEmpty() ? 0 : Integer.parseInt(countString));
        });

        binding.addCigarettesCancelButton.setOnClickListener(view -> dismiss());
    }
}

File: app/src/main/java/com/madebyaron/betterhealth/view/list/dialogs/AddCigarettesDialog.kt
package com.madebyaron.betterhealth.view.list.dialogs

import android.app.Dialog
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.ViewModelProviders
import com.madebyaron.betterhealth.R
import com.madebyaron.betterhealth.view.components.DateTimePicker
import com.madebyaron.betterhealth.viewmodel.ListModel
import kotlinx.android.synthetic.main.dialog_add_cigarettes.*
import java.util.*

class AddCigarettesDialog(private val activity: FragmentActivity) : Dialog(activity) {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.dialog_add_cigarettes)
		window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))

		val model = ViewModelProviders.of(activity).get(ListModel::class.java)
		var date = Date()

		DateTimePicker.setDateTimePickerText(add_cigarettes_date_input, date)
		add_cigarettes_date_input.setOnClickListener {
			DateTimePicker.openDateTimePickerDialog(context, date) { dateTime ->
				date = dateTime
				DateTimePicker.setDateTimePickerText(add_cigarettes_date_input, date)
			}
		}

		add_cigarettes_add_button.setOnClickListener {
			dismiss()
			val countString = add_cigarettes_count_input.text.toString()
			model.addCigarettes(date,
					if (countString.isBlank()) {
						0
					} else {
						countString.toInt()
					}
			)
		}

		add_cigarettes_cancel_button.setOnClickListener {
			dismiss()
		}
	}
}